fastparquet/writer.py

```diff
@@ -571,6 +574,12 @@ def write(filename, data, partitions=[0], encoding="PLAIN",
                     fmd.row_groups.append(rg)
             fmd.num_rows = sum(rg.num_rows for rg in fmd.row_groups)
         else:
+            if write_index or write_index is None and not (
```

Comment:
How about higher at the top of `write()`. This makes it simpler because we don't have to special case metadata or pass down kwargs through the call chain.